### PR TITLE
[unimodules] Remove experience references

### LINKED ISSUE #3517: Permissions depend on experienceId
expo-permissions currently expects an experienceId to exist, but outside of the Expo context it is nil. Consequently, when a user accepts a permission we get this crash:

```
GrowlerProwler[87996:4885419] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '*** -[__NSDictionaryM setObject:forKeyedSubscript:]: key cannot be nil'
*** First throw call stack:
(
	0   CoreFoundation                      0x00000001074ae1bb __exceptionPreprocess + 331
	1   libobjc.A.dylib                     0x0000000105bcf735 objc_exception_throw + 48
	2   CoreFoundation                      0x00000001073fa4ec _CFThrowFormattedException + 194
	3   CoreFoundation                      0x00000001075207ac -[__NSDictionaryM setObject:forKeyedSubscript:] + 1036
	4   GrowlerProwler                      0x0000000100860a72 -[EXPermissionsManager savePermission:ofType:forExperience:] + 802
	5   GrowlerProwler                      0x000000010085c0a8 __67-[EXPermissions askForGlobalPermissions:withResolver:withRejecter:]_block_invoke.95 + 440
	6   GrowlerProwler                      0x000000010085a347 -[EXLocationRequester locationManager:didChangeAuthorizationStatus:] + 183
	7   CoreLocation                        0x0000000102e9ed74 CLClientStopVehicleHeadingUpdates + 71531
	8   CoreLocation                        0x0000000102e9ea15 CLClientStopVehicleHeadingUpdates + 70668
	9   CoreLocation                        0x0000000102e89210 CLClientInvalidate + 1351
	10  CoreFoundation                      0x000000010741362c __CFRUNLOOP_IS_CALLING_OUT_TO_A_BLOCK__ + 12
	11  CoreFoundation                      0x0000000107412de0 __CFRunLoopDoBlocks + 336
	12  CoreFoundation                      0x000000010740dc2b __CFRunLoopRun + 2779
	13  CoreFoundation                      0x000000010740ce11 CFRunLoopRunSpecific + 625
	14  GraphicsServices                    0x000000010c7b81dd GSEventRunModal + 62
	15  UIKitCore                           0x0000000110cee81d UIApplicationMain + 140
	16  GrowlerProwler                      0x000000010080e240 main + 112
	17  libdyld.dylib                       0x0000000108f79575 start + 1
	18  ???                                 0x0000000000000001 0x0 + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
``` 

This can be resolved by initializing to a stub experienceId or making each method handle a nil experienceId. I believe in this case the offending line that can't handle a nil value is: `    experiencePermissions = [[NSMutableDictionary alloc] initWithDictionary:expoPermissions[experienceIdKey]];` - however there could be others so we'd need to a do a proper passthrough.

### LINKED ISSUE #3864: [expo-av] Bare workflow app crashes on use of Audio methods
## 🐛 Bug Report

Upon running the following in a **bare** workflow app:

```js
const soundObject = new Audio.Sound();
await soundObject.loadAsync({ uri: audio });
await soundObject.playAsync();
```

An error occurs (from [the line of code here](https://github.com/expo/expo/blob/429f1245387adbb5dcc56e3315927467d4a34c2e/packages/expo-av/ios/EXAV/EXAudioSessionManager.m#L256)):

`Experience requesting Audio Session changes has no ID`

Followed by `Play encountered an error: audio session not activated.`

I guess this is because bare apps don't have an `experienceId`?

Is it possible to add a temporary workaround à la this issue: https://github.com/expo/expo/issues/3517? 🙂

### PR DESCRIPTION
**Pull request by sjchmiela**
# Why

Properly closes https://github.com/expo/expo/pull/3875, fixes https://github.com/expo/expo/issues/3517, fixes https://github.com/expo/expo/issues/3864.

# How

Removed all `experience` references from universal modules, adding scoped implementations to ExpoKit.

# Test Plan

- [x] NCL works
- [x] bare template works

### CODE REVIEW

**Review by bbarthec — APPROVED** [review]

**Review by tsapeta — APPROVED** [review]
lgtm 👍

File: ios/Exponent/Versioned/Core/UniversalModules/EXScopedFileSystem/EXScopedFilePermissionModule.m [hunk 1]
```diff
@@ -15,8 +22,7 @@ - (UMFileSystemPermissionFlags)getExternalPathPermissions:(NSString *)path
 }
 
 - (BOOL)shouldForbidAccessToExternalDirectories {
-  id<UMConstantsInterface> constantsModule = [[self moduleRegistry] getModuleImplementingProtocol:@protocol(UMConstantsInterface)];
-  return [@"expo" isEqualToString:constantsModule.appOwnership];
+  return ![EXEnvironment sharedEnvironment].isDetached;
```
Comment (sjchmiela) [inline ios/Exponent/Versioned/Core/UniversalModules/EXScopedFileSystem/EXScopedFilePermissionModule.m]: Can't use `EXEnvironment here`

**Comment by naorzr** [comment]
I'm using the latest rn-unimodules(0.3.1), and the problem of #3875 persists for me, is this pr still hadn't been released?

**Comment by tsapeta** [comment]
That's right, these changes haven't been released yet. They will be out once we release SDK33 😉

**Pull request merged**